ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixParticipant.java

```diff
@@ -91,6 +94,22 @@ public void initialize(String hostName, int port, List<AmbryHealthReport> ambryH
     }
   }
 
+  @Override
+  public void setReplicaSealedState(ReplicaId replicaId, boolean isSealed) {
+    HelixAdmin helixAdmin = manager.getClusterManagmentTool();
+    InstanceConfig instanceConfig = helixAdmin.getInstanceConfig(clusterName, instanceName);
+    List<String> sealedReplicas = ClusterMapUtils.getSealedReplicas(instanceConfig);
+    if (sealedReplicas == null)
+      sealedReplicas = new ArrayList<>();
+    if (isSealed) {
+      sealedReplicas.add(replicaId.getPartitionId().toPathString());
```

Comment:
what if it already exists? Does helix dedup? If not, perhaps a exists check might be useful.

Reply:
Is the sealedReplicas list gonna be pretty small?  What's the expected scale?

Reply:
*Since de-duping on a list will scale with the list size

Reply:
There's already a "remove" in there which makes it linear anyway... I'll add the de-duper, but maybe we should look into replacing the list with a set